fe/fe-core/src/main/java/org/apache/doris/ldap/LdapManager.java

```diff
@@ -200,24 +207,22 @@ private LdapUserInfo getUserInfoFromCache(String fullName) {
      * Step2: get roles by ldap groups;
      * Step3: merge the roles;
      */
-    private Role getLdapGroupsPrivs(String userName, String clusterName) throws DdlException {
+    private Set<Role> getLdapGroupsRoles(String userName, String clusterName) throws DdlException {
         //get user ldap group. the ldap group name should be the same as the doris role name
         List<String> ldapGroups = ldapClient.getGroups(userName);
-        List<String> rolesNames = Lists.newArrayList();
+        Set<Role> roles = Sets.newHashSet();
         for (String group : ldapGroups) {
             String qualifiedRole = ClusterNamespace.getFullName(clusterName, group);
             if (Env.getCurrentEnv().getAuth().doesRoleExist(qualifiedRole)) {
-                rolesNames.add(qualifiedRole);
+                roles.add(Env.getCurrentEnv().getAuth().getRoleByName(qualifiedRole));
             }
         }
-        LOG.debug("get user:{} ldap groups:{} and doris roles:{}", userName, ldapGroups, rolesNames);
+        LOG.debug("get user:{} ldap groups:{} and doris roles:{}", userName, ldapGroups, roles);
 
         Role ldapGroupsPrivs = new Role(LDAP_GROUPS_PRIVS_NAME);
```

Comment:
LDAP_GROUPS_PRIVS_NAME renamed to `defaultPrivs` or `defaultRole` or something else would be better, it can no longer represent the meaning of ldapGroupsPrivs. `infomation_schema select_priv` is also the default privilege of doris user, you can consider creating such a role to `RoleManager`, so that ldap user can save only the role name like doris user.

Reply:
if create a default role in RoleManager,user will can grant priv to this role,that will change auth of all ldap user.so I change the name if LDAP_GROUPS_PRIVS_NAME to ldapDefaultRole only.